README.md

```diff
@@ -136,59 +143,133 @@ only the request being passed to the middleware.
 #### 4.2.1 Projects Using Single Pass
 
 There are fewer examples of this approach within projects using HTTP Messages,
-with a couple of notable exceptions.
+with one notable exception.
 
-[Guzzle middleware](http://docs.guzzlephp.org/en/latest/handlers-and-middleware.html)
-uses a rather unique approach to middleware that is based around generators and
-promises. Ignoring the generator portion, Guzzle middleware has the signature:
+[Guzzle middleware][guzzle-middleware] is focused on outgoing (client) requests
+and uses this signature:
 
 ```
 function(RequestInterface $request, array $options): ResponseInterface
 ```
 
-[Laravel middleware](https://laravel.com/docs/master/middleware) does make use
-of HTTP Messages but supports middleware with the signature:
+#### 4.2.2 Additional Projects Using Single Pass
+
+There are also significant projects that predate HTTP Messages using this approach.
+
+[StackPHP][stackphp] is based on [Symfony HttpKernel][httpkernel] and supports
+middleware with this signature:
 
 ```
-function handle(Request $request, callable $next): Response
+handle(Request $request, $type, $catch): Response
```

Comment:
This contradicts line 119, where you indicated that the StackPHP "style" is `fn(request, frame): response`. I'm thinking that the narrative at line 119 should likely not mention Stack at all.